src/eth1spec/base_types.py

```diff
@@ -0,0 +1,553 @@
+
+    def __rdivmod__(self, left: int) -> Tuple["U256", "U256"]:
+        if not isinstance(left, int):
+            return NotImplemented
+
+        if left < 0 or left > self.MAX_VALUE:
+            raise ValueError()
+
+        result = super(U256, self).__rdivmod__(left)
+        return (self.__class__(result[0]), self.__class__(result[1]))
+
+    def unchecked_pow(self, right: int, modulo: Optional[int] = None) -> int:
+        if modulo is not None:
+            if not isinstance(modulo, int):
+                return NotImplemented
+
+            if modulo < 0 or modulo > self.MAX_VALUE:
+                raise ValueError()
+
+        if not isinstance(right, int):
+            return NotImplemented
+
+        if right < 0 or right > self.MAX_VALUE:
+            raise ValueError()
+
+        return super(U256, self).__pow__(right, modulo)
+
+    def wrapping_pow(self, right: int, modulo: Optional[int] = None) -> "U256":
+        result = self.unchecked_pow(right, modulo)
+
+        if result == NotImplemented:
+            return NotImplemented
+
+        result %= 2 ** 256
+        return self.__class__(result)
+
+    def __pow__(self, right: int, modulo: Optional[int] = None) -> "U256":
+        result = self.unchecked_pow(right, modulo)
+
+        if result == NotImplemented:
+            return NotImplemented
+
+        return self.__class__(result)
+
+    def __rpow__(self, left: int, modulo: Optional[int] = None) -> "U256":
+        if modulo is not None:
+            if not isinstance(modulo, int):
+                return NotImplemented
+
+            if modulo < 0 or modulo > self.MAX_VALUE:
+                raise ValueError()
+
+        if not isinstance(left, int):
+            return NotImplemented
+
+        if left < 0 or left > self.MAX_VALUE:
+            raise ValueError()
+
+        result = super(U256, self).__rpow__(left, modulo)
+        return self.__class__(result)
+
+    def __ipow__(self, right: int, modulo: Optional[int] = None) -> "U256":
+        return self.__pow__(right, modulo)
+
+    # TODO: Implement and, or, xor, neg, pos, abs, invert, ...
+
+    def to_be_bytes32(self) -> "Bytes32":
+        """
+        Converts this 256-bit unsigned integer into its big endian
+        representation with exactly 32 bytes.
+
+        Returns
+        -------
+        big_endian : `Bytes32`
+            Big endian (most significant bits first) representation.
+        """
+        return self.to_bytes(32, "big")
+
+    def to_be_bytes(self) -> "Bytes":
+        """
+        Converts this 256-bit unsigned integer into its big endian
+        representation, omitting leading zero bytes.
+
+        Returns
+        -------
+        big_endian : `Bytes`
+            Big endian (most significant bits first) representation.
+        """
+        bit_length = self.bit_length()
+        byte_length = (bit_length + 7) // 8
+        return self.to_bytes(byte_length, "big")
+
+
+U256.MAX_VALUE = U256(2 ** 256 - 1)
+
+
+Bytes = bytes
```

Comment:
Any reason this doesn't use [`typing.NewType`](https://docs.python.org/3/library/typing.html#newtype)?

Reply:
We haven't finished the type annotations yet :sweat_smile: